src/ucp/core/ucp_request.c

```diff
@@ -317,22 +317,19 @@ int ucp_request_pending_add(ucp_request_t *req)
 static unsigned ucp_request_dt_invalidate_progress(void *arg)
 {
     ucp_request_t *req = arg;
-    ucp_ep_h ucp_ep    = req->send.ep;
 
     ucp_request_complete_send(req, req->status);
-    ucp_ep_refcount_remove(ucp_ep, invalidate);
-
     return 1;
 }
 
 static void ucp_request_mem_invalidate_completion(uct_completion_t *comp)
 {
     ucp_request_t *req         = ucs_container_of(comp, ucp_request_t,
                                                   send.state.uct_comp);
-    ucp_ep_h ep                = req->send.ep;
+    ucp_worker_h worker        = req->send.worker;
```

Comment:
no need to keep worker in request.
Can do:
`ucp_worker_h worker = ucs_container_of(ucs_mpool_obj_owner(req), ucp_worker_t, req_mp);`

Reply:
it doesn't work if request is a user one

Reply:
how about `req->send.invalidate.worker` ?

Reply:
done